DigitalOffice/Tests/ProjectServiceTests/Repositories/EditProjectByIdRepositoryTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+using Microsoft.EntityFrameworkCore;
+using LT.DigitalOffice.Kernel.UnitTestLibrary;
+using LT.DigitalOffice.ProjectService.Database;
+using LT.DigitalOffice.ProjectService.Repositories;
+using LT.DigitalOffice.ProjectService.Database.Entities;
+
+namespace LT.DigitalOffice.ProjectServiceUnitTests.Repositories
+{
+    public class EditProjectByIdRepositoryTests
+    {
+        private Guid departmentId;
+
+        private DbProject existingProject;
+        private DbProject editProject;
+
+        private ProjectRepository repository;
+        private ProjectServiceDbContext dbContext;
+        private DbContextOptions<ProjectServiceDbContext> dbOptionsProjectService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dbOptionsProjectService = new DbContextOptionsBuilder<ProjectServiceDbContext>()
+                .UseInMemoryDatabase(databaseName: "ProjectServiceTest")
+                .Options;
+
+            dbContext = new ProjectServiceDbContext(dbOptionsProjectService);
+            repository = new ProjectRepository(dbContext);
+
+            departmentId = Guid.NewGuid();
+
+            existingProject = new DbProject
+            {
+                Name = "A test name",
+                DepartmentId = departmentId,
+                Description = "Description",
+                IsActive = true,
+                Deferred = false
+            };
+
+            editProject = new DbProject
+            {
+                Name = "Is different",
+                DepartmentId = Guid.NewGuid(),
+                Description = "Is different too",
+                IsActive = false,
+                Deferred = false
+            };
+        }
+
+        [Test]
+        public void SuccessfulEditProjectTest()
+        {
+            existingProject.Id = Guid.NewGuid();
+            editProject.Id = existingProject.Id;
+
+            dbContext.Add(existingProject);
+            dbContext.SaveChanges();
+            dbContext.Entry(existingProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var dbProject = dbContext.Projects
```

Comment:
I do not understand for what purpose this variable exists. You already have existingProject

Reply:
existingProject is a project that is definetily added to the database. Later it's assigned to a new variable dbProject.

You know, I should swap their names so that's clearer to understand.
So I'll first add dbProject to Db and then get an existingProject from it.

Reply:
Fixed. You can have a look.